libs/core-ui/src/lib/DatasetCohort.ts

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) Microsoft Corporation.
```

Comment:
This PR is really large and hard to review.  Maybe we can break it up a bit more - eg the DatasetCohort.ts could be its own PR to start with.